skimage/io/_io.py

```diff
@@ -49,15 +59,42 @@ def imread(fname, as_gray=False, plugin=None, **plugin_args):
         RGB-image MxNx3 and an RGBA-image MxNx4.
 
     """
-    if isinstance(fname, pathlib.Path):
-        fname = str(fname.resolve())
+    if plugin_args:
+        warnings.warn(
+            "Additional keyword arguments are deprecated since version "
+            "0.23 and will be removed in 0.25 (or later). To avoid this "
+            "warning, please do not use additional keyword arguments. "
+            "Use imageio or a similar package instead.",
+            category=FutureWarning,
+            stacklevel=2,
+        )
 
-    if plugin is None and hasattr(fname, 'lower'):
-        if fname.lower().endswith(('.tiff', '.tif')):
-            plugin = 'tifffile'
+    with warnings.catch_warnings():
+        warnings.filterwarnings(
+            action="ignore",
+            message=".*Use imageio or a similar package instead.*",
+            category=FutureWarning,
+            module="skimage.io",
+        )
+        use_plugin = (
+            plugin is not DEPRECATED
+            or plugin_args
+            or plugin_order()["imread"][0] != "imageio"
+        )
+        if use_plugin:
```

Comment:
Because we want a deprecation period during which we support both APIs (our old one and imageio), right? 

I'm probably overthinking this. Let me look into this again and try for a simpler approach.